### PR TITLE
Refactor DescribeIndex

### PR DESCRIPTION
**Pull request by boraarslan**

### CODE REVIEW

File: quickwit/quickwit-cli/src/index.rs [hunk 1]
```diff
@@ -643,122 +645,199 @@ pub async fn describe_index_cli(args: DescribeIndexArgs) -> anyhow::Result<()> {
     let metastore = metastore_uri_resolver
         .resolve(&quickwit_config.metastore_uri)
         .await?;
-    let index_metadata = metastore.index_metadata(&args.index_id).await?;
-    let splits = metastore
-        .list_splits(&args.index_id, SplitState::Published, None, None)
-        .await?;
+    let index_stats = IndexStats::from_metastore(metastore, &args.index_id).await?;
```
Comment (guilload) [inline quickwit/quickwit-cli/src/index.rs:648]: I'd get the index metadata and splits outside of this function and then change the signature to `IndexStats::new(index_metadata, splits)`.

**Comment by guilload** [comment]
Let's also test the essential functions of the two new stats structs.

File: quickwit/quickwit-cli/tests/helpers.rs [hunk 2]
```diff
@@ -228,3 +229,16 @@ pub fn create_test_env(index_id: String, storage_type: TestStorageType) -> anyho
         storage,
     })
 }
+
```
Comment (guilload) [inline quickwit/quickwit-cli/tests/helpers.rs:232]: `#[cfg(any(test, feature = "testsuite"))]`

File: quickwit/quickwit-cli/tests/cli.rs [hunk 3]
```diff
@@ -1073,3 +1078,85 @@ async fn test_cmd_all_with_s3_localstack_internal_api() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_index_stats() -> anyhow::Result<()> {
+    let index_id = append_random_suffix("index-stats-env");
```
Comment (guilload) [inline quickwit/quickwit-cli/tests/cli.rs:1084]: You don't have to randomize those if they're not going to be written to a shared test resource (LocalStack, Kafka, ...)

File: quickwit/quickwit-cli/src/index.rs [hunk 4]
```diff
@@ -643,122 +645,222 @@ pub async fn describe_index_cli(args: DescribeIndexArgs) -> anyhow::Result<()> {
...
+            .map(|split| (split.split_metadata.footer_offsets.end / 1_000_000) as usize)
+            .sorted()
+            .collect_vec();
+        let total_bytes = splits_bytes.iter().sum::<usize>();
+
+        let timestamp_range = if index_metadata.indexing_settings.timestamp_field.is_some() {
+            let time_min = splits
+                .iter()
+                .flat_map(|split| split.split_metadata.time_range.clone())
+                .map(|time_range| *time_range.start().min(time_range.end()))
```
Comment (guilload) [inline quickwit/quickwit-cli/src/index.rs:737]: Why do we need `time_range.start().min(time_range.end()`? Is `time_range.start() <= time_range.end()` not always true?

Comment (boraarslan) [reply quickwit/quickwit-cli/src/index.rs]: `start >= end` is possible for `RangeInclusive<T>` but it should not be possible for `SplitMetadata` you are right. Fixed.

**Review by guilload — APPROVED** [review]

**Pull request merged**